tsfresh/feature_extraction/extraction.py

```diff
@@ -190,32 +195,68 @@ def _extract_features_parallel_per_sample(kind_to_df_map, settings, column_id, c
                                                            column_value=column_value,
                                                            settings=settings)
     pool = Pool(settings.n_processes)
+    total_number_of_expected_results = 0
 
     # Submit map jobs per kind per sample
     results_fifo = Queue()
+
     for kind, df_kind in kind_to_df_map.items():
         df_grouped_by_id = df_kind.groupby(column_id)
+
+        total_number_of_expected_results += len(df_grouped_by_id)
+
+        chunksize = _calculate_best_chunksize(df_grouped_by_id, settings)
+
         results_fifo.put(
-            pool.map_async(
+            pool.imap(
```

Comment:
I have no idea. @nils-braun Wouldn't it better to use an imap_unordered? The ordering shouldn't matter to us and I'd suspect that unordered is faster.

I'll measure the performance of using the imap vs the map_async. If it is noticeably slower, we could revert to map_async and provide a less granular progress.